### PR TITLE
Added unit tests for cursor moving through editor

### PR DESCRIPTION
**Pull request by BogdanPaul15**
This pull request adds some unit tests for cursor moving.
Issue: [#124 ](https://github.com/ilai-deutel/kibi/issues/124)

### CODE REVIEW

File: src/editor.rs [hunk 1]
```diff
@@ -813,8 +813,9 @@ impl PromptMode {
                 PromptState::Completed(b) => {
                     let mut args = b.split_whitespace();
                     match Command::new(args.next().unwrap_or_default()).args(args).output() {
-                        Ok(out) if !out.status.success() =>
-                            set_status!(ed, "{}", String::from_utf8_lossy(&out.stderr).trim_end()),
+                        Ok(out) if !out.status.success() => {
+                            set_status!(ed, "{}", String::from_utf8_lossy(&out.stderr).trim_end());
+                        }
```
Comment (ilai-deutel) [inline src/editor.rs:818]: Can you revert these lines? The curly braces are not necessary here and add an extra line

```suggestion
                        Ok(out) if !out.status.success() =>
                            set_status!(ed, "{}", String::from_utf8_lossy(&out.stderr).trim_end()),
```

File: src/editor.rs [hunk 2]
```diff
@@ -924,4 +925,164 @@ mod tests {
         editor.delete_char();
         assert_eq!(editor.rows[0].chars, "Helo world".as_bytes());
     }
+
+    #[test]
+    fn editor_move_cursor_left() {
+        let mut editor = Editor::default();
+        for b in "Hello world!\nHappy New Year!".as_bytes() {
```
Comment (ilai-deutel) [inline src/editor.rs:932]: Nit: `"Hello world!\nHappy New Year!".as_bytes()` can be simplified as `b"Hello world!\nHappy New Year!"` (and similarly below)

**Review by ilai-deutel — COMMENTED** [review]
Thank you very much for this PR! Looks like this [increases the total code coverage by 7.28%](https://app.codecov.io/gh/ilai-deutel/kibi/pull/270)!

**Comment by ilai-deutel** [comment]
@all-contributors please add @BogdanPaul15 for test

File: src/editor.rs (hunk 2 above)
Comment (BogdanPaul15) [reply src/editor.rs:932]: Thank you for the feedback. I will commit these changes soon.

**Pull request merged**